script.js

```diff
@@ -98,3 +101,41 @@ function getData(featuredImage, live, source) {
   popUpImage.innerHTML = result;
   linkLive.innerHTML = link;
 }
+
+// form validation
+const form = document.querySelector('.form-view');
+
+function validForm(e) {
+  e.preventDefault();
+  const regex = /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$/;
+  if (emailValue.value.match(regex)) {
+    form.submit();
+    return true;
+  }
+  errorMsg.innerHTML = 'please email is invalid';
+  return false;
+}
+
+document.querySelector('.form-view').addEventListener('submit', validForm);
+
+function storeData() {
+  const formData = {
+    name: nameValue.value,
+    email: emailValue.value,
+    message: msgValue.value,
+  };
+  localStorage.setItem('formdata', JSON.stringify(formData));
+}
+
+document.querySelector('.form-view').addEventListener('submit', storeData);
+
+function loadData() {
+  const userData = JSON.parse(localStorage.getItem('formdata') || '[]');
+  if (userData !== 'undefined' || userData !== null) {
+    nameValue.value = userData.name;
+    emailValue.value = userData.email;
+    msgValue.value = userData.message;
+  }
+}
```

Comment:
- [ ] As you may notice how the prefilled values of your inputs are `undefined` when the page load. Please make the right data are saved or retrieved from your `localStorage.` as you may see the string value `'[ ]'` instead if there is nothing from the storage which there can be no way it has some values like `userData.name`,` ...`
YOu may consider deleting that `'[ ]'` and also say something like nameValue = userData.name || `" "`
![image](https://github.com/ctokoli/personal-portfolio-site/assets/97100091/b0b58c65-51cc-45ac-a6e3-1ba1b9891906)